src/Microsoft.XmlSerializer.Generator/src/Sgen.cs

```diff
@@ -341,13 +341,20 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
         // assumes all same case.        
         private bool ArgumentMatch(string arg, string formal)
         {
-            if (arg[0] != '/' && arg[0] != '-')
+            if (arg[0] != '-' || arg.Length==1)
             {
                 return false;
             }
-
-            arg = arg.Substring(1);
-            return (arg == formal || (arg.Length == 1 && arg[0] == formal[0]));
+            if (arg[1] == '-')
```

Comment:
We need check if arg.Length is bigger than 1 first.

Reply:
@huanwu The check is in line 344